packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -402,6 +444,88 @@ export class NatGateway extends Resource implements IRouteTarget {
   }
 }
 
+/**
+ * Creates a peering connection between two VPCs
+ * @resource AWS::EC2::VPCPeeringConnection
+ */
+export class VPCPeeringConnection extends Resource implements IRouteTarget {
+
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerTargetId: string;
+
+  /**
+   * The VPC peering connection CFN resource.
+   */
+  public readonly resource: CfnVPCPeeringConnection;
+
+  constructor(scope: Construct, id: string, props: VPCPeeringConnectionProps) {
+    super(scope, id);
+
+    const region = props.acceptorRegion ? props.acceptorRegion : Stack.of(this).region;
+    let peerRole: Role | undefined;
+    this.routerType = RouterType.VPC_PEERING_CONNECTION;
+
+    if (props.isCrossAccount && props.acceptorAccountId === undefined) {
+      throw new Error('AcceptorAccountId is required for cross-account peering connections');
+    }
+
+    const overlap = validateVpcCidrOverlap(props.requestorVpc, props.acceptorVpc);
+    if (overlap) {
+      throw new Error('CIDR block should not overlap with existing subnet blocks');
+    }
+
+    if (props.isCrossAccount) {
+      peerRole = new Role(this, 'PeerRole', {
+        assumedBy: new AccountPrincipal(Stack.of(this).account),
+        roleName: `VPCPeeringRole-${id}`,
+      });
+
+      peerRole.assumeRolePolicy?.addStatements(
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ['sts:AssumeRole'],
+          principals: [new AccountPrincipal(Stack.of(this).account)],
+        }),
+      );
+
+      peerRole.addToPolicy(new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: ['ec2:acceptVpcPeeringConnection'],
+        resources: [`arn:${Aws.PARTITION}:ec2:${region}:${props.acceptorAccountId}:vpc/${props.acceptorVpc.vpcId}`],
+      }));
+
+      peerRole.addToPolicy(new PolicyStatement({
+        actions: ['ec2:acceptVpcPeeringConnection'],
+        effect: Effect.ALLOW,
+        resources: [`arn:${Aws.PARTITION}:ec2:${region}:${props.acceptorAccountId}:vpc-peering-connection/*`],
+        conditions: {
+          StringEquals: {
+            'ec2:AccepterVpc': `arn:${Aws.PARTITION}:ec2:${region}:${props.acceptorAccountId}:vpc/${props.acceptorVpc.vpcId}`,
+          },
+        },
+      }));
+    }
```

Comment:
Could you point me to the docs that specify these policies?

Reply:
The highly restrictive policy is what we're implementing from here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/peer-with-vpc-in-another-account.html#create-template-with-highly-restrictive-policy